src/python_inspector/resolution.py

```diff
@@ -501,12 +507,45 @@ def format_resolution(results, environment, repos, as_tree=False):
         return dependencies
 
 
+def pdt_dfs(mapping, graph, src):
+    """
+    Return a nested mapping of dependencies.
```

Comment:
Can you elaborate on what the inputs are and what is the processing taking place here?